### ISSUE TITLE
About the alpha for fade-in effect

### ISSUE THREAD

**Issue by deartonym** [akanimax/pro_gan_pytorch#2]
In the process of training, there is a fade-in parameter called alpha for both Generator loss gen_loss and Discriminator loss dis_loss, which value is calculated at the beginning of every epoch:

```
            # calculate the value of aplha for fade-in effect
            alpha = int(epoch / num_epochs)
```
However, this alpha will always be 0 except the last epoch. It seems very weird and this fade-in effect is not discussed in the original paper.

Is there any explanation for this? Thanks!

**Comment by akanimax** [comment]
Hi @deartonym,

The `cifar-10` code displayed on the `README` for the package is merely a sample for explaining the Interface of the package. For more detailed example, refer to the **`LFW`** experiment of [pro_gan_pytorch-examples](https://github.com/akanimax/pro_gan_pytorch-examples). 

About setting `alpha = int(epoch / num_epochs)` at the beginning of the epoch, this is not integer point division `a // b`, but floating point division. so the value of alpha changes after every epoch for the fade-in effect.

Please refer to `Section 2: paragraph 2` 

> All existing layers in both networks remain trainable throughout the training process. When new layers are added to the networks, we fade them in smoothly, as illustrated in Figure 2.

and `Figure 2` of the paper for the concept of fade-in effect. This is fully complaint to the specifications of the paper.

I hope this helps,

Cheers :beers:!
@akanimax

**Comment by deartonym** [comment]
Thanks for your explanation!

Though '/' is a floating division, there is an 'int' after the division which will always make the alpha into 0 except the last epoch.

I misunderstood the Fig. 2 and thought that the fading parameter was fixed at 0.5 and was discreet instead of a continuously decreasing parameter!

Thanks again.

Best,
Shu

**Comment by deartonym** [comment]
Though '/' is a floating division, there is an 'int' after the division which will always make the alpha into 0 except the last epoch.

Thanks again.

**Comment by akanimax** [comment]
@deartonym,
Aaah! yes. Fixed in [5715ef1a09e32f96edd0d6ae075a791663b09edc](https://github.com/akanimax/pro_gan_pytorch/commit/5715ef1a09e32f96edd0d6ae075a791663b09edc).
Thanks!

Best regards,
@akanimax

**Comment by deartonym** [comment]
@akanimax No problem Aka!

Best,
Shu
